woocommerce/payment-gateway/class-sv-wc-payment-gateway-payment-form.php

```diff
@@ -1013,7 +1032,27 @@ public function render_fieldset_end() {
 	 */
 	public function render_js() {
 
-		wc_enqueue_js( $this->get_safe_handler_js() );
+		// skip if script already rendered
+		if ( true === $this->js_rendered ) {
+			return;
+		}
+
+		$safe_handler_js = '<script type="text/javascript">jQuery(function($) { ' . $this->get_safe_handler_js() . ' });</script>';
```

Comment:
`wc_enqueue_js()` would push this to be loaded in footer anyway, because `wc_print_js()` hooks to `wp_footer`, so I am not sure if I understand the benefit of doing the same thing here - furthermore `wc_print_js` hooks a little later at priority `25`, I am not sure why but could it be so it ensures `jQuery` is enqueued in case it's added in the footer?

what is exactly the advantage of having this hook callback and what problem is trying to solve? the mechanism is not clear to me

Reply:
maybe check my PR review comment, I think there could be a case to still use `wc_enqueue_js` somehow

Reply:
considering that some plugins may override `render_js` as you pointed elsewhere, how about calling `$this->render_js()` after those checks? if the problem is the hook context, how about using something a bit higher than `wp_footer` but which is ensured to be present in the relevant pages we are interested in?

Reply:
The problem with using `wc_enqueue_js` doesn't work due to the same reason you mentioned, the `render_js` gets called when the `wp_footer` action is triggered, it it's matter of race condition.

I tried to offset this by adjusting the `priority` argument to ensure the callback gets called before the `wc_print_js` but with no luck, `add_action( 'wp_footer', [ $this, 'load_render_js' ], 0 );` I am not sure why it doesn't work